gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -190,75 +195,102 @@ var gpii = fluid.registerNamespace("gpii");
-                },
-                makeSnapshot: function (handlerResponse) {
-                    // update the settings section of our snapshot to contain the new information
-                    var settingsSnapshot = gpii.lifecycleManager.responseToSnapshot(solutionId, handlerResponse);
-                    // Settings handlers may or may not return options (currently it seems they all do) - gain resistance to this by restoring the
-                    // original "options" supplied to them.
-                    fluid.each(handlerSpec, function (entry, key) {
-                        if (key !== "settings") {
-                            settingsSnapshot[key] = fluid.copy(entry);
-                        }
-                    });
-                    return settingsSnapshot;
-                }
-            };
-        });
-        var responsePromise = fluid.promise.sequence(fluid.getMembers(settingsPackage, "setSettings"));
+    gpii.lifecycleManager.invokeSettingsHandlers = function (that, solutionId, handlerSpec) {
+        // first prepare the payload for the settingsHandler in question - a more efficient
+        // implementation might bulk together payloads destined for the same handler
+        var settingsHandlerPayload = gpii.lifecycleManager.specToSettingsHandler(solutionId, handlerSpec);
+        var resolvedName = that.nameResolver.resolveName(handlerSpec.type, "settingsHandler");
+        var setSettingsPromise = gpii.settingsHandlers.dispatchSettingsHandler(resolvedName, settingsHandlerPayload, that.options.retryOptions);
         var togo = fluid.promise();
-        responsePromise.then(function (responses) {
-            var snapshots = fluid.transform(responses, function (handlerResponse, i) {
-                return settingsPackage[i].makeSnapshot(handlerResponse);
+
+        setSettingsPromise.then(function (handlerResponse) {
+            // update the settings section of our snapshot to contain the new information
+            var settingsSnapshot = gpii.lifecycleManager.responseToSnapshot(solutionId, handlerResponse);
+            // Settings handlers may or may not return options (currently it seems they all do) - gain resistance to this by restoring the
+            // original "options" supplied to them.
+            fluid.each(handlerSpec, function (entry, key) {
+                if (key !== "settings") {
+                    settingsSnapshot[key] = fluid.copy(entry);
+                }
             });
-            togo.resolve(snapshots);
+            togo.resolve(settingsSnapshot);
         }, togo.reject);
+
         return togo;
     };
 
+
     gpii.lifecycleManager.invokeAction = function (action, nameResolver) {
         var resolvedName = nameResolver.resolveName(action.type, "action");
         return fluid.invokeGradedFunction(resolvedName, action);
     };
 
+    /**
+     * @param action {String} should be either a reference to a settings block, written as settings.<blockName>
```

Comment:
This docs comment is now hugely out of date. Also should add @return {Promise} describing what is returned